frontend/packages/shared/src/hooks/queries/useResourcePolicySubjectsQuery.ts

```diff
@@ -0,0 +1,41 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { useServicesContext } from 'app-shared/contexts/ServicesContext';
+import type { PolicySubject } from '@altinn/policy-editor';
+import { QueryKey } from 'app-shared/types/QueryKey';
+import { AxiosError } from 'axios';
+
+const policySubjectOrg: PolicySubject = {
+  subjectDescription: '[org]',
+  subjectId: '[org]',
+  subjectSource: '[org]',
+  subjectTitle: '[org]',
+};
+
+/**
+ * Query to get the list of subjects for a policy.
+ *
+ * @param org the organisation of the user
+ * @param repo the repo the user is in
+ * @param addOrgToList if the value for org should be added to the list of subjects
+ *
+ * @returns UseQueryResult with a list of subjects of PolicySubject
+ */
+export const useResourcePolicySubjectsQuery = (
+  org: string,
+  repo: string,
+  addOrgToList?: boolean,
+): UseQueryResult<PolicySubject[], AxiosError> => {
+  const { getPolicySubjects } = useServicesContext();
+
+  return useQuery<PolicySubject[], AxiosError>(
+    [QueryKey.ResourcePolicySubjects, org, repo],
+    () => getPolicySubjects(org, repo),
+    {
+      onSuccess: (data: PolicySubject[]) => {
+        if (addOrgToList && !data.some((d) => d.subjectId === policySubjectOrg.subjectId))
+          data.push(policySubjectOrg);
+        return data;
+      },
+    },
+  );
```

Comment:
Hva tenker du om å endre navn fra "data" til "policySubjects" for at det skal bli mer beskrivende hva det faktisk er, uten å måtte se på TS-typen? 😊 Hva tenker du om å returnere en kopi av responsen inn i if-setningen istedenfor å pushe data på responsen som kommer fra serveren? 😊 

```suggestion
  return useQuery<PolicySubject[], AxiosError>(
    [QueryKey.ResourcePolicySubjects, org, repo],
    () => getPolicySubjects(org, repo),
    {
      onSuccess: (policySubjects: PolicySubject[]) => {
        if (addOrgToList && !policySubjects.some((d) => d.subjectId === policySubjectOrg.subjectId)) {
              return [....policySubjects, policySubjectOrg];
          }
        
        return policySubjects;
      },
    },
  );
```